spl/token/_layouts.py

```diff
@@ -0,0 +1,73 @@
+"""Token instruction layouts."""
+from enum import IntEnum
+
+from construct import Int8ul, Int64ul, Pass  # type: ignore
+from construct import Struct as cStruct
+from construct import Switch
+
+from solana._layouts.shared import PUBLIC_KEY_LAYOUT
+
+
+class InstructionType(IntEnum):
+    """Token instruction types."""
+
+    InitializeMint = 0
+    InitializeAccount = 1
+    InitializeMultisig = 2
+    Transfer = 3
+    Approve = 4
+    Revoke = 5
+    SetAuthority = 6
+    MintTo = 7
+    Burn = 8
+    CloseAccount = 9
+    FreezeAccount = 10
+    ThawAccount = 11
+    Transfer2 = 12
+    Approve2 = 13
+    MintTo2 = 14
+    Burn2 = 15
```

Comment:
what is the different between `Transfer` and `Transfer2`? and so on?

Reply:
`Transfer2` allows the caller to specify the mint address and decimals.